packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -432,40 +454,144 @@ export class DbTargetGlobal {
+    { unique: true }
+  );
+  db.createObjectStore(DbTargetGlobal.store);
+}
+
 /**
- * The list of all IndexedDB stored used by the SDK. This is used when creating
- * transactions so that access across all stores is done atomically.
+ * An object representing the changes at a particular snapshot version for the
+ * given target. This is used to facilitate storing query changelogs in the
+ * targetChanges object store.
+ *
+ * PORTING NOTE: This is used for change propagation during multi-tab syncing
+ * and not needed on iOS and Android.
  */
-export const ALL_STORES = [
+export class DbTargetChange {
+  /** Name of the IndexedDb object store.  */
+  static store = 'targetChanges';
+
+  /** Keys are automatically assigned via the targetId and snapshotVersion. */
+  static keyPath = ['targetId', 'snapshotVersion'];
+
+  constructor(
+    /**
+     * @param targetId - The targetId identifying a target.
+     */
+    public targetId: TargetId,
+    /**
+     * @param snapshotVersion - The snapshot version for this change.
+     */
+    public snapshotVersion: DbTimestamp,
+    /**
+     * @param changes - The keys of the changed documents in this snapshot.
+     */
+    public changes: {
+      added?: EncodedResourcePath[];
+      modified?: EncodedResourcePath[];
+      removed?: EncodedResourcePath[];
+    }
+  ) {}
+}
+
+function createTargetChangeStore(db: IDBDatabase): void {
+  db.createObjectStore(DbTargetChange.store, {
+    keyPath: DbTargetChange.keyPath as KeyPath
+  });
+}
+
+/**
+ * A record of the metadata state of each instance.
+ *
+ * PORTING NOTE: This is used for primary-tab selection to support multi-tab
+ * persistence and does not need to be ported to iOS or Android.
+ */
+export class DbInstanceMetadata {
+  /** Name of the IndexedDb object store. */
+  static store = 'instanceMetadata';
+
+  /** Keys are automatically assigned via the userId and instanceKey properties. */
+  static keyPath = ['userId', 'instanceKey'];
+
+  constructor(
+    /** @param userId - The normalized user ID to which this batch belongs. */
+    public userId: string,
+    /**
+     * @param  instanceKey - The auto-generated instance key assigned at client
+     * startup.
+     */
+    public instanceKey: string,
+    /** @param  updateTimeMs - The last time this state was updated. */
+    public updateTimeMs: DbTimestamp
+  ) {}
+}
+
+function createInstanceMetadataStore(db: IDBDatabase): void {
+  db.createObjectStore(DbInstanceMetadata.store, {
+    keyPath: DbInstanceMetadata.keyPath as KeyPath
+  });
+}
+
+// Exported for testing.
+export const V1_STORES = [
+  DbDocumentMutation.store,
   DbMutationQueue.store,
   DbMutationBatch.store,
-  DbDocumentMutation.store,
-  DbRemoteDocument.store,
-  DbTarget.store,
   DbOwner.store,
+  DbRemoteDocument.store,
+  DbTargetDocument.store,
   DbTargetGlobal.store,
-  DbTargetDocument.store
+  DbTarget.store
 ];
+
+const SCHEMA_V2_STORES = [DbInstanceMetadata.store, DbTargetChange.store];
```

Comment:
V1_STORES, SCHEMA_V2_STORES, ALL_STORES

One of these is not like the others.

Reply:
Fixed